examples/checkpointing/checkpoint.py

```diff
@@ -0,0 +1,65 @@
+from pyrevolve import Checkpoint, Operator
+from devito import TimeFunction
+
+
+class DevitoOperator(Operator):
```

Comment:
Why not a fist-citizen class in ``operator.py`` ? 

A name such as ``CheckpointOperator`` would also be much, much more informative

Reply:
Other question (also based on what I read in ``apply.__doc__``). Does pyrevolve depend on Devito? I guess no -- in fact, it should be the other way round. So how can pyrevolve know about ``Operators`` and ``apply``? Or am I getting this completely wrong?

Reply:
> Does pyrevolve depend on Devito? 
No
> So how can pyrevolve know about Operators and apply? 
pyRevolve provides an `Operator` ABC that has an apply method. This is a concrete implementation of the `pyrevolve.Operator` class that wraps the `devito.Operator` class.